web/models/class/class.ex

```diff
@@ -82,4 +85,27 @@ defmodule CoursePlanner.Class do
   end
   def validate_offered_course(changeset), do: changeset
 
+  defp validate_date(%{valid?: true, changes: %{date: date, offered_course_id: oc_id}} = changeset) do
+    term = OfferedCourse
+    |> Repo.get(oc_id)
+    |> Repo.preload([:term])
+    |> Map.get(:term)
+
+    st = term
+    |> Map.get(:start_date)
+    |> Date.cast!()
+
+    en = term
+    |> Map.get(:end_date)
+    |> Date.cast!()
+
+    case {Date.compare(st, date), Date.compare(en, date)} do
+      {:gt, _} -> Changeset.add_error(changeset, :date,
+                  "The class date is before term's beginning")
```

Comment:
could you remove `The class date ` from the error message?

when this is added beside the field with the error tag in the page it doesn't read as nice